TestMetricsDuGetCapacity unit test fails on SLES 12 Go 1.8 ppc64le on master branch

<!-- Thanks for filing an issue! Before hitting the button, please answer these questions.-->

**Is this a request for help?** (If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.):


**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.):
TestMetricsDuGetCapacity
---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT
<!--
If this is a BUG REPORT, please:
  - Fill in as much of the template below as you can.  If you leave out
    information, we can't help you as well.

If this is a FEATURE REQUEST, please:
  - Describe *in detail* the feature/behavior/change you'd like to see.

In both cases, be ready for followup questions, and please respond in a timely
manner.  If we can't reproduce a bug or think a feature already exists, we
might close your issue.  If we're wrong, PLEASE feel free to reopen it and
explain why.
-->

**Kubernetes version** (use `kubectl version`):
master branch (tip)

**Environment**:
- **Cloud provider or hardware configuration**:
- **OS** (e.g. from /etc/os-release):
SLES 12
- **Kernel** (e.g. `uname -a`):
Linux pts00449-vm23 3.12.28-4-default #1 SMP Thu Sep 25 17:02:34 UTC 2014 (9879bd4) ppc64le ppc64le ppc64le GNU/Linux
- **Install tools**:
- **Others**:


**What happened**:
make test WHAT="./pkg/volume"
Running tests for APIVersion: v1,apps/v1beta1,authentication.k8s.io/v1,authentication.k8s.io/v1beta1,authorization.k8s.io/v1,authorization.k8s.io/v1beta1,autoscaling/v1,autoscaling/v2alpha1,batch/v1,batch/v2alpha1,certificates.k8s.io/v1beta1,extensions/v1beta1,imagepolicy.k8s.io/v1alpha1,policy/v1beta1,rbac.authorization.k8s.io/v1beta1,rbac.authorization.k8s.io/v1alpha1,settings.k8s.io/v1alpha1,storage.k8s.io/v1beta1,storage.k8s.io/v1,federation/v1beta1
+++ [0404 11:22:46] Running tests without code coverage
--- FAIL: TestMetricsDuGetCapacity (0.01s)
        metrics_du_test.go:73: Unexpected Used for directory with file.  Expected 4096, got 65536.
FAIL
FAIL    k8s.io/kubernetes/pkg/volume    0.137s
Makefile:164: recipe for target 'test' failed
make: *** [test] Error 1

Test case failed
**What you expected to happen**:
Test case should pass

**How to reproduce it** (as minimally and precisely as possible):
make test WHAT="./pkg/volume"

**Anything else we need to know**:
Found this similar issue: https://github.com/kubernetes/kubernetes/issues/20028
However the values returned are different, see this string "Expected 4096, got 65536"

--- a/pkg/volume/metrics_du_test.go
+++ b/pkg/volume/metrics_du_test.go
@@ -23,13 +23,22 @@ import (
 	"os"
 	"path/filepath"
 	"testing"
+	"syscall"
 
 	utiltesting "k8s.io/client-go/util/testing"
 	. "k8s.io/kubernetes/pkg/volume"
 	volumetest "k8s.io/kubernetes/pkg/volume/testing"
 )
 
-const expectedBlockSize = 4096
+func getExpectedBlockSize(path string) (int64) {
+	statfs := &syscall.Statfs_t{}
+	err := syscall.Statfs(path, statfs)
+	if err != nil {
+		return 0
+	}
+
+	return int64(statfs.Bsize)
+}
 
 // TestMetricsDuGetCapacity tests that MetricsDu can read disk usage
 // for path
@@ -69,7 +78,7 @@ func TestMetricsDuGetCapacity(t *testing.T) {
 	if err != nil {
 		t.Errorf("Unexpected error when calling GetMetrics %v", err)
 	}
-	if e, a := (expectedEmptyDirUsage.Value() + expectedBlockSize), actual.Used.Value(); e != a {
+	if e, a := (expectedEmptyDirUsage.Value() + getExpectedBlockSize(filepath.Join(tmpDir, "f1"))), actual.Used.Value(); e != a {
 		t.Errorf("Unexpected Used for directory with file.  Expected %v, got %d.", e, a)
 	}
 }

--- a/pkg/volume/metrics_du_test.go
+++ b/pkg/volume/metrics_du_test.go
@@ -22,15 +22,15 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
-	"testing"
 	"syscall"
+	"testing"
 
 	utiltesting "k8s.io/client-go/util/testing"
 	. "k8s.io/kubernetes/pkg/volume"
 	volumetest "k8s.io/kubernetes/pkg/volume/testing"
 )
 
-func getExpectedBlockSize(path string) (int64) {
+func getExpectedBlockSize(path string) int64 {
 	statfs := &syscall.Statfs_t{}
 	err := syscall.Statfs(path, statfs)
 	if err != nil {

